Tasks/Maven/maventask.ts

```diff
@@ -227,14 +227,43 @@ function applyPmdGoals(mvnRun: trm.ToolRunner):void {
 }
 
 // Returns true if any one code analysis tool was enabled.
-function isCodeAnalysisReportRequired():boolean {
+function isCodeAnalysisUploadRequired():boolean {
     return tl.getInput('pmdAnalysisEnabled', true) == 'true';
 }
 
+// Take a result object and attempt to upload the XML and HTML reports as artifacts. Throws if fail.
+function uploadBuildArtifactsFromResult(analysisResult:ar.AnalysisResult):void {
+    // make sure the staging folder exists and is empty
+    tl.rmRF(stagingDir);
+    tl.mkdirP(stagingDir);
```

Comment:
Using a the agent staging directory (cleaned between builds) as a temporary folder. This is a workaround required because the Linux agent has a problem with uploading individual files at this time, and can only upload entire folders at once.